sdk/src/test/java/io/dapr/client/DaprHttpTest.java

```diff
@@ -177,16 +179,48 @@ public void invokePostMethodUnknownError() throws IOException {
 
     @Test
     public void getHeadersAndStatus(){
-
         mockInterceptor.addRule()
                 .post("http://localhost:3500/v1.0/state")
                 .respond(500, ResponseBody.create(MediaType.parse("application/json"),
                         "{\"errorCode\":\"null\",\"message\":\"null\"}"));
-
         DaprHttp daprHttp = new DaprHttp(3500, okHttpClient);
-
         System.out.println(daprHttp);
+    }
 
+    @Test ()
+    public void testCallbackCalledAtTheExpectedTimeTest() throws IOException {
```

Comment:
Can you add a comment describing what this is validating and how?  Another way to look at this is "why this test is needed"?

Ideally unit tests self-documenting by name and from being short, but this one is more complicated.

Reply:
Documentation for the case has been added.

Reply:
Thanks for adding documentation.
"Block" in the comments here makes it sound like it is stopping the call from happening (e.g. blocking the network).  You could change the "Block" on 194, 198, 199 to "Block for" or "Wait for".

I'm asking to write down the scenario this method aims to test, because sometimes you see test cases that seem very specific or particular and the reason it was added is lost with time.  For example, it may not be clear to a reader why `responseDeleted` and `responseDeleteKey` are created in one order, then .block() invoked on them in the opposite order.  And why is the `addRule()` on line 227 added after resetting the interceptor?  These both must be to orchestrate a specific scenario - a description of what that is should go above the test method.

Reply:
@LMWF can you look at the current documentation?

Reply:
Thanks for expanding.